src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -390,8 +390,54 @@ public static String getExternalServiceUrl(OpenShiftClient openShiftClient, Stri
 	 * @return the external URL to access Jenkins
 	 */
 	public static String getJenkinsURL(OpenShiftClient openShiftClient, String namespace) {
+        // if the user has explicitly configured the jenkins root URL, use it
+        String rootUrl = Jenkins.getInstance().getRootUrl();
+        if (StringUtils.isNotEmpty(rootUrl)) {
+            return rootUrl;
+        }
+
+        // otherwise, we'll see if we are running in a pod and can infer it from the service/route
+        //TODO we will eventually make the service name configurable, with the default of "jenkins"
 		return getExternalServiceUrl(openShiftClient, "http://", namespace, "jenkins");
 	}
+	
+	public static String getNamespacefromPodInputs() {
```

Comment:
i think this should cache the value (or should only be called once, during plugin initialization, rather than per build update as is currently the case).  The pod's namespace can't change.

Reply:
good catch